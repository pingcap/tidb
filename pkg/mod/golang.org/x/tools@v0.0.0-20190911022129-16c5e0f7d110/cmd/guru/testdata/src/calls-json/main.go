package main

// Tests of call-graph queries, -format=json.
// See go.tools/guru/guru_test.go for explanation.
// See calls-json.golden for expected query results.

func call(f func()) {
	f() // @callees @callees-f "f"
}

func main() {
	call(func() {
		// @callers callers-main.anon "^"
		// @callstack callstack-main.anon "^"
	})
}
