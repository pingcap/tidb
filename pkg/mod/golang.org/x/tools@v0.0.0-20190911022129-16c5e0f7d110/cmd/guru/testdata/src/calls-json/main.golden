-------- @callees @callees-f --------
{
	"pos": "testdata/src/calls-json/main.go:8:3",
	"desc": "dynamic function call",
	"callees": [
		{
			"name": "calls-json.main$1",
			"pos": "testdata/src/calls-json/main.go:12:7"
		}
	]
}
-------- @callstack callstack-main.anon --------
{
	"pos": "testdata/src/calls-json/main.go:12:7",
	"target": "calls-json.main$1",
	"callers": [
		{
			"pos": "testdata/src/calls-json/main.go:8:3",
			"desc": "dynamic function call",
			"caller": "calls-json.call"
		},
		{
			"pos": "testdata/src/calls-json/main.go:12:6",
			"desc": "static function call",
			"caller": "calls-json.main"
		}
	]
}
